model/aberrations.js

```diff
@@ -64,66 +71,185 @@ exports.upsertAber = function(data, callback){
 	}
 	if (err) {
-            console.log("Error upserting gene annotation: " + err);
-	    console.log("Debug: full query:", query.string)
-	    callback(err, null)
+            console.log("Error upserting gene aberration: " + err);
+	    callback(err, null);
+	    return
 	}
     }
 
-    // todo: test update case
-    // todo: transaction-ize w/ rollback (not necessary, just good to clean the annos table)
-    // TODO: may insert duplicates, fix this if we want something more interesting with genes
+    // TODO: we may insert duplicate mutations, fix this if we want something more interesting with genes
 
+    var now = new Date().toString().substring(0,15);
+    // parse the protein sequence change
+    var acids = data.protein_seq_change.split(/\d+/), locus = 0; 
+    if (acids[0].length < data.protein_seq_change.length)
+	locus = parseInt(data.protein_seq_change.substr(acids[0].length));
+    if (acids.length == 1)
+	acids[1] = '?';
+    
     var aberInsertQuery = abers.insert(
 	abers.gene.value(data.gene),
-	abers.transcript.value(data.transcript),
-	abers.mut_class.value(data.mut_class),
-	abers.mut_type.value(data.mut_type),
-	abers.protein_seq_change.value(data.protein_seq_change),
-	abers.aber_user_id.value(data.user_id)).returning(abers.u_id);
-
-    Database.execute(aberInsertQuery, function(err, subresult) {
-	handleErr(err, subresult, aberInsertQuery);
-	var aber_u_id = subresult.rows[0].u_id;
-	data.aber_id = aber_u_id;
-	console.log(data);
-	exports.upsertSourceAnno(data, callback);
-    });
+	abers.mutation_class.value(data.mut_class),
+	abers.mutation_type.value(data.mut_type),
+	abers.locus.value(locus),
+	abers.original_amino_acid.value(acids[0]),
+	abers.new_amino_acid.value(acids[1]),
+	abers.last_edited.value(now),
+	abers.created_on.value(now)).returning(abers.id);
+
+	DjangoDatabase.execute(aberInsertQuery, function(err, subresult) {
+	    handleErr(err, subresult, aberInsertQuery);
+	    var aber_u_id = subresult.rows[0].id;
+	    data.aber_id = aber_u_id;
+	    exports.upsertSourceAnno(data, callback);
+	});
+}
+
+// callback supplies user data
+function getUserInfo(user, callback) {
+
+    DjangoDatabase.execute(
+	Schemas.users.where({'email': user.email}),
+	function (err, user_subresult) {
+	    if(err) 
+		callback(err, null);
+	    else if (user_subresult.rows.length == 1) 
+		callback(null, user_subresult.rows[0]);
+	    else if (user_subresult.rows.length == 0) {
+		first_last_names = user.name.split(" ");
+		if (first_last_names.length == 1) 
+		    first_last_names[1] = '';
+
+		// todo: pass up error with error requesting login 
+		DjangoDatabase.execute(
+		    Schemas.users.insert({'first_name': first_last_names[0],
+				  'last_name': first_last_names[1],
+				  'email': user.email,
+				  'username': user.email.split('@')[0],
+				  'is_staff': false,
+				  'is_superuser': false,
+				  'is_active': false,
+				  'date_joined': getTime("day"),
+				  'password': 'deadbeef'}).returning('*'),
```

Comment:
This field is the password for the user on the MAGI django annotations website, which is separate from the server's authentication to communicate with the Django database. 

The password in the Django database is a hash that Django somehow constructs - I'm not sure how it interacts with Google OAuth.  A preferred behavior would be to ask the user to login on the Django portion of the website so that Django can learn about the user on its own.